src/pubtools/sign/signers/cosignsigner.py

```diff
@@ -219,9 +238,50 @@ def container_sign(self: CosignSigner, operation: ContainerSignOperation):
         signing_results.operation_result = operation_result
         return signing_results
 
+    def existing_signatures(self, reference: str) -> Tuple[bool, str]:
+        """Return list of existing signatures for given reference.
+
+        Args:
+            reference (str): reference to get list of signatures for
+        Returns:
+            Tuple[bool, str]: tuple of success flag and error message or result string
+        """
+        common_args = [
+            self.cosign_bin,
+            "-t",
+            self.timeout,
+            "triangulate",
```

Comment:
I may be misremembering, but I don't know if triangulate command actually checks for the image's existence.

Reply:
I'm not sure what you mean. You mean it doesn't check if repository exist first?

Reply:
I seem to recall that this command only returns the image reference of the signature image without taking into account if the image actually exists. But I may be misremembering, so if you're confident that the command works as expected, then that's good enough for me.

Reply:
if image or tag don't exist it returns error

Reply:
It doesn't check existence of signature.
```
# cosign triangulate quay.io/redhat-dev/test----repo114@sha256:6c55558d600c9f5462ca25102dadba3f0e3724cf665770b0283bd83014857c18
quay.io/redhat-dev/test----repo114:sha256-6c55558d600c9f5462ca25102dadba3f0e3724cf665770b0283bd83014857c18.sig

# skopeo inspect docker://quay.io/redhat-dev/test----repo114:sha256-6c55558d600c9f5462ca25102dadba3f0e3724cf665770b0283bd83014857c18.sig
FATA[0000] Error parsing image name "docker://quay.io/redhat-dev/test----repo114:sha256-6c55558d600c9f5462ca25102dadba3f0e3724cf665770b0283bd83014857c18.sig": reading manifest sha256-6c55558d600c9f5462ca25102dadba3f0e3724cf665770b0283bd83014857c18.sig in quay.io/redhat-dev/test----repo114: manifest unknown 
```